homeassistant/components/wemo/wemo_device.py

```diff
@@ -81,11 +81,26 @@ async def _async_subscription_callback(self, updated: bool) -> None:
         else:
             self.async_set_updated_data(None)
 
+    @property
+    def should_poll(self) -> bool:
+        """Return True if polling is needed to update the state for the device.
+
+        The alternative, when this returns False, is to rely on the subscription
+        "push updates" to update the device state in Home Assistant.
+        """
+        if isinstance(self.wemo, Insight) and self.wemo.get_state() == 0:
```

Comment:
Are we sure that `wemo.get_state` won't do I/O here?

Reply:
Calling get_state() or get_state(force_update=False) on a pyWeMo device instance is async-safe after the first call. wemo_device.async_register_device does an initial call. From that point forward these calls are async-safe.